RevenueCatUI/Helpers/TrialOrIntroEligibilityChecker.swift

```diff
@@ -43,6 +43,25 @@ extension TrialOrIntroEligibilityChecker {
         return await self.eligibility(for: package.storeProduct)
     }
 
+    /// Computes eligibility for a list of packages in parallel, returning them all in a dictionary.
+    func eligibility(for packages: [Package]) async -> [Package: IntroEligibilityStatus] {
+        return await withTaskGroup(of: (Package, IntroEligibilityStatus).self) { group in
+            for package in packages {
+                group.addTask {
+                    return (package, await self.eligibility(for: package))
```

Comment:
why though? we allow for batch computation on eligibility

Reply:
this really should be a thing the main SDK does instead - it already can take in a list of product ids, it should take in a list of packages and map the eligibility with them. 
Otherwise this could be unnecessarily heavy

Reply:
I forgot to mention that. I started refactoring it and it was a pain because we only have an API for identifiers, so we need to reconstruct a `[Package: IntroEligibilityStatus]` after getting the result, which was a lot of boilerplate without access to all our fancy `Set` and `Dictionary` extensions...

But I agree it's important, I'll change this.

Reply:
Added a new API in #2858, and optimized this using it in #2846.

Reply:
(To be fair this won't optimize SK2 because there's no bulk API, but it will for SK1).